app/src/main/java/app/revanced/integrations/settingsmenu/ReturnYouTubeDislikeSettingsFragment.java

```diff
@@ -80,6 +89,19 @@ public void onCreate(Bundle savedInstanceState) {
         });
         preferenceScreen.addPreference(percentagePreference);
 
+        compactLayoutPreference = new SwitchPreference(context);
```

Comment:
Preference screens such as this originate from legacy code by VancedTeam. Unrelated to this PR, but they should be converted to ReVanced style of settings.

Reply:
What should this be migrated to?

Reply:
The patch should use the `settings` patch as a dependency and integrate its settings into the app. The `settings` patch does not support "custom" preference components yet, which would be necessary in the case of the `sponsorblock` patch.

Reply:
It would need to handle the statistics info that is shown/not shown based on debug status, and the additional RYD switches that become non selectable when  RYD is not enabled (percentage, compact layout).

Reply:
Yes, the `settings` patch would need to take a custom transformer which patches can supply if necessary. Not relevant to this PR though, just something to consider.